nullaway/src/main/java/com/uber/nullaway/fixserialization/Serializer.java

```diff
@@ -213,9 +220,99 @@ public static String serializeSymbol(@Nullable Symbol symbol) {
+            .map(
+                parameter -> {
+                  // varargs is only defined on the method and not on the parameter. This
+                  // information is lost while visiting the last parameter in the method if it is
+                  // varargs and the parameter is visited as a normal array type. We need to
+                  // manually convert the last parameter to varargs if the method is varargs.
+                  Type type =
+                      // if the parameter is the last parameter,
+                      (index.getAndIncrement() == methodSymbol.getParameters().size()
+                              // if the method is varargs,
+                              && methodSymbol.isVarArgs()
+                              // for safety of casting, check if the parameter is an array type
+                              && parameter.type instanceof Type.ArrayType)
+                          ? ((Type.ArrayType) parameter.type).makeVarargs()
+                          : parameter.type;
+                  return type.accept(
+                      new Types.DefaultTypeVisitor<String, Void>() {
+                        @Override
+                        public String visitWildcardType(Type.WildcardType t, Void unused) {
+                          StringBuilder sb = new StringBuilder();
+                          sb.append(t.kind);
+                          if (t.kind != BoundKind.UNBOUND) {
+                            sb.append(t.type.accept(this, null));
+                          }
+                          return sb.toString();
+                        }
+
+                        @Override
+                        public String visitCapturedType(Type.CapturedType t, Void s) {
+                          return t.wildcard.accept(this, null);
+                        }
+
+                        @Override
+                        public String visitClassType(Type.ClassType t, Void unused) {
+                          StringBuilder sb = new StringBuilder();
+                          sb.append(t.tsym);
+                          if (t.getTypeArguments().nonEmpty()) {
+                            sb.append('<');
+                            sb.append(
+                                t.getTypeArguments().stream()
+                                    .map(arg -> arg.accept(this, null))
+                                    .collect(joining(",")));
+                            sb.append(">");
+                          }
+                          return sb.toString();
+                        }
+
+                        @Override
+                        public String visitType(Type t, Void unused) {
+                          return t.toString();
+                        }
+
+                        @Override
+                        public String visitArrayType(Type.ArrayType t, Void unused) {
+                          return t.elemtype.accept(this, null) + (t.isVarargs() ? "..." : "[]");
```

Comment:
It's nice to print varargs as `...` but it shouldn't be necessary to uniquely identify a method.  Do we need this?  If not we could simplify our code a bit

Reply:
It is required to work with existing logic with injector and javaparser. But, I agree that it is better to have a simpler logic in NullAway and update injector to work with either `...` and `[]`. Please let me spend some time on injector and update it and then resolve remaining comments, as some are related to this. Thank you.

Reply:
@msridhar Ok, so I updated injector and now it can work with either `...` or `[]`. Removed this logic. d74cde16a2d44e830d1b5caa52d1efdb9ba3edd6